src/main/java/ch/njol/skript/expressions/ExprKeptItems.java

```diff
@@ -0,0 +1,147 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import java.util.List;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.inventory.ItemStack;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Kept Items")
+@Description({
+	"The kept items in a 'death of player' event.",
+	"Note: this expression won't remove the items from the drops when changed.",
+	"Use the <a href='effects.html#EffKeepItems'>keep items</a> effect for that."
+})
+@Examples({
+	"on death of player:",
+		"\tset kept items to the drops where [input isn't victim's tool]"
+})
+@Events("death")
+@RequiredPlugins("Paper")
+@Since("INSERT VERSION")
+public class ExprKeptItems extends SimpleExpression<ItemType> {
+
+	static {
+		if (Skript.methodExists(PlayerDeathEvent.class, "getItemsToKeep"))
+			Skript.registerExpression(ExprKeptItems.class, ItemType.class, ExpressionType.SIMPLE, "[the] kept items");
```

Comment:
Maybe I didn't explain it clearly enough in my last review, this syntax is really ambiguous.

The difference between what is dropped when the player dies, what doesn't drop and what is given to them after respawning is very murky and needs to be made explicit.
I think an entirely different name for this syntax would help.